packages/material-ui/src/Divider/Divider.js

```diff
@@ -1,181 +1,252 @@
+        borderLeft: 0,
+        borderRight: 0,
+        borderBottom: 0,
+        borderStyle: 'solid',
+        top: '50%',
+        content: '""',
+        transform: 'translateY(50%)',
+      },
+    }),
+    /* Styles applied to the root element if divider have text and `orientation="vertical"`. */
+    ...(styleProps.children &&
+      styleProps.orientation === 'vertical' && {
+        flexDirection: 'column',
+        '&::before, &::after': {
+          height: '100%',
+          top: '0%',
+          left: '50%',
+          borderColor: theme.palette.divider,
+          borderTop: 0,
+          borderLeft: 'thin',
+          borderStyle: 'solid',
+          transform: 'translateX(0%)',
+        },
+      }),
+  }),
+  ({ styleProps }) => ({
+    /* Styles applied to the root element if `textAlign="right" orientation="horizontal"`. */
+    ...(styleProps.textAlign === 'right' &&
+      styleProps.orientation !== 'vertical' && {
+        '&::before': {
+          width: '90%',
+        },
+        '&::after': {
+          width: '10%',
+        },
+      }),
+    /* Styles applied to the root element if `textAlign="left" orientation="horizontal"`. */
+    ...(styleProps.textAlign === 'left' &&
+      styleProps.orientation !== 'vertical' && {
+        '&::before': {
+          width: '10%',
+        },
+        '&::after': {
+          width: '90%',
+        },
+      }),
+  }),
+);
+
+const DividerWrapper = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiDivider',
+    slot: 'Wrapper',
   },
-  /* Styles applied to the span children element if `orientation="horizontal"`. */
-  wrapper: {
+)(
+  ({ theme }) => ({
     display: 'inline-block',
     paddingLeft: theme.spacing(1.2),
     paddingRight: theme.spacing(1.2),
-  },
-  /* Styles applied to the span children element if `orientation="vertical"`. */
-  wrapperVertical: {
-    paddingTop: theme.spacing(1.2),
-    paddingBottom: theme.spacing(1.2),
-  },
-});
+  }),
+  ({ theme, styleProps }) => ({
+    ...(styleProps.orientation === 'vertical' && {
+      paddingTop: theme.spacing(1.2),
+      paddingBottom: theme.spacing(1.2),
+    }),
+  }),
+);
 
 const Divider = React.forwardRef(function Divider(props, ref) {
   const {
     absolute = false,
-    classes,
     className,
     children,
-    component: Component = children ? 'div' : 'hr',
+    component = children ? 'div' : 'hr',
     flexItem = false,
     light = false,
     orientation = 'horizontal',
-    role = Component !== 'hr' ? 'separator' : undefined,
+    role = component !== 'hr' ? 'separator' : undefined,
     textAlign = 'center',
     variant = 'fullWidth',
     ...other
   } = props;
 
-  const themeVariantsClasses = useThemeVariants(
-    {
-      ...props,
-      absolute,
-      component: Component,
-      flexItem,
-      light,
-      orientation,
-      role,
-      textAlign,
-      variant,
-    },
-    'MuiDivider',
-  );
+  const styleProps = {
+    ...props,
+    absolute,
+    children,
```

Comment:
```suggestion
    component,
```

Reply:
The `children` prop directly affects the styles too, though, so shouldn't we leave it in `styleProps`, separated from the spread `...props`?

Reply:
It doesn’t have default value and it is already in the props :) the component however has a default value so we have to specify it

Reply:
Ahhhh, so that's the difference -- gotcha, thanks for the info!